### ISSUE TITLE
PHP 7.2: count(): Parameter must be an array or an object that implements Countable

### ISSUE THREAD

**Issue by vasilvestre** [schmittjoh/JMSSerializerBundle#688]
Similar to this issue : https://github.com/yiisoft/yii/issues/4167

The default behaviour of count changed, it now fire a warning. It happen for example here : https://github.com/schmittjoh/serializer/blob/1.x/src/JMS/Serializer/XmlDeserializationVisitor.php#L179

```diff
- if (!\count($nodes)) {
+ if ($nodes === false || !\count($nodes)) {
```
Or whatever condition we choose, to keep the default Serializer behaviour the check is needed

**Comment by goetas** [comment]
This issue is not related to PHP 7.2 and neither is a bug.
It is more related to DX. 

In this case the solution is to throw an exception in `$nodes === false` and probably in the error message mentioning the value of `$entryName`.

**Comment by vasilvestre** [comment]
(Link to yii was just to give another case of the problem in another case)

So if it's false, an error have to be thrown ?

**Comment by goetas** [comment]
> So if it's false, an error have to be thrown ?

yes. there is no right "default" behavior the lib could do except of throwing an error.

**Comment by vasilvestre** [comment]
I don't know, it's supposed to throw an error if the node is empty ? Isn't it a problem of code logic or content ?

**Comment by goetas** [comment]
Do not know it the error was thrown for that reason... I think the error was in `$entryName`.. wasn't?

**Comment by vasilvestre** [comment]
There's no contributing guide in this repo, I would like to help on this one but I need something to follow

**Comment by goetas** [comment]
You have created the ticker in the wrong repo :) (thats why I have added the "serializer issue" label)

The ticket shoild have been opened on https://github.com/schmittjoh/serializer.
Into that library you have https://github.com/schmittjoh/serializer/blob/master/CONTRIBUTING.md as contributing guide

**Comment by goetas** [comment]
Please note that most likely you will have to submit your PR to the `1.x` branch.
